core/templates/services/loader.service.ts

```diff
@@ -34,22 +32,22 @@ export class LoaderService {
    * In order to keep the variables same, static is used until migration is
    * complete.
    */
-  static loadingMessageSubject: Subject<string> = new Subject<string>();
+  static loadingMessageChangedEventEmitter = new EventEmitter<string>();
   constructor() {}
 
   showLoadingScreen(message: string): void {
     // TODO(#9154): Change LoaderService to "this".
-    LoaderService.loadingMessageSubject.next(message);
+    LoaderService.loadingMessageChangedEventEmitter.emit(message);
   }
 
   hideLoadingScreen(): void {
     // TODO(#9154): Change LoaderService to "this".
-    LoaderService.loadingMessageSubject.next('');
+    LoaderService.loadingMessageChangedEventEmitter.emit('');
   }
 
-  getLoadingMessageSubject(): Subject<string> {
+  onLoadingMessageChange(): EventEmitter<string> {
```

Comment:
I just realized it's semantically a bit weird for onXXX to return something. Normally stuff like onresize just does an action after the resize event happens. But this is not called when the loading message is changed, right?

So I think I'm a bit confused about what semantically this is supposed to mean. Could you explain? It seems more like a  getLoadingMessageChangedEventEmitter().

Reply:
Yeah, +1 on this.